### PR TITLE
installation: testing support for PyPy

### PR DESCRIPTION
**Pull request by jirikuncar**

### CODE REVIEW

**Comment by tiborsimko** [comment]
Are all our dependencies PyPy friendly already?  If not, we'd be introducing N new unnecessary CI jobs in the testing matrix, which is kind of going against the spirit of https://github.com/inveniosoftware/invenio/issues/3647.

Note that that PyPy testing seems to take 2x more time on Travis when compared to CPython: https://travis-ci.org/inveniosoftware/invenio-config/builds/144410515

**Comment by jirikuncar** [comment]
> Note that that PyPy testing seems to take 2x more time on Travis when compared to CPython

The pip cache was not initialized.

> Are all our dependencies PyPy friendly already?

I'm trying to find out by starting with *base* packages.

**Comment by tiborsimko** [comment]
Can you repush with `allow_failures` for Travis CI?

File: .travis.yml [hunk 1]
```diff
@@ -41,6 +41,11 @@ env:
 python:
   - "2.7"
   - "3.5"
+  - "pypy"
+
+matrix:
+  allow_failures:
```
Comment (jirikuncar) [inline .travis.yml]: ping @tiborsimko

Comment (tiborsimko) [inline .travis.yml]: Merged.  BTW shouldn't we add also: 

- `Programming Language :: Python :: Implementation :: CPython` 

in addition to:

-  `Programming Language :: Python :: Implementation :: PyPy`

?

**Pull request merged**